src/lib/Repository/UiLocationService.php

```diff
@@ -52,14 +63,38 @@ public function loadLocations(ContentInfo $contentInfo)
         return $uiLocations;
     }
 
+    /**
+     * Deletes locations.
+     * Throws exception if there is an attempt to delete the main location.
+     *
+     * @param array $locationIds
+     * @param int $contentId
+     *
+     * @throws ForbiddenException
+     */
+    public function deleteLocations(array $locationIds, int $contentId)
+    {
+        $contentInfo = $this->contentService->loadContentInfo($contentId);
+
+        foreach ($locationIds as $locationId) {
+            $location = $this->locationService->loadLocation($locationId);
+
+            if ($this->isMainLocation($location, $contentInfo)) {
+                throw new ForbiddenException('Main location cannot be deleted.');
+            }
+
+            $this->locationService->deleteLocation($location);
+        }
+    }
+
     private function buildUiLocations(array $locations, ContentInfo $contentInfo)
     {
         return array_map(
             function (Location $location) use ($contentInfo) {
                 $properties = [
                     'childCount' => $this->locationService->getLocationChildCount($location),
                     'pathLocations' => $this->pathService->loadPathLocations($location),
-                    'main' => ($location->id === $contentInfo->mainLocationId),
+                    'main' => $this->isMainLocation($location, $contentInfo),
```

Comment:
`isMain` maybe ? and maybe UiLocation should have a `isMain` method instead of injecting that?